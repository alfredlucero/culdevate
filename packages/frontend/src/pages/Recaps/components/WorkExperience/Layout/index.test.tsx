import React from "react";
import { render } from "@testing-library/react";
import WorkExperienceLayout, { WorkExperienceLayoutProps } from "./index";

const defaultProps: WorkExperienceLayoutProps = {
  recaps: [
    {
      kind: "Work Experience",
      userId: "userId",
      _id: "workExperienceId",
      bulletPoints: [
        "Worked on the Culdevate start up seriously",
        "Getting better at making REST API endpoints",
        "Improving interviewing and mentoring skills",
      ],
      startDate: new Date("2020/01/17").toISOString(),
      title: "Lead Software Engineer",
      company: "Culdevate",
      location: "Long Beach, CA",
      employmentType: "Self-Employed",
    },
  ],
  onGoBackToLanding: () => {},
  onCreateRecapSuccess: () => {},
  onUpdateRecapSuccess: () => {},
  onDeleteRecapSuccess: () => {},
};

describe("<WorkExperienceLayout />", () => {
  test("should render without error", () => {
    const { container } = render(
      <WorkExperienceLayout
        {...defaultProps}
        testId="workExperienceLayoutTestId"
        className="extra-work-experience-layout-class"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should show the empty state when there are no recaps", () => {
    const { getByTestId, queryByTestId } = render(<WorkExperienceLayout {...defaultProps} recaps={[]} />);

    expect(queryByTestId("workExperienceRecap")).toBeNull();

    expect(getByTestId("workExperienceEmptyCard")).toBeVisible();
  });

  test("should show the recaps when there is data", () => {
    const { getByTestId, queryByTestId } = render(<WorkExperienceLayout {...defaultProps} />);

    expect(queryByTestId("workExperienceEmptyCard")).toBeNull();

    expect(getByTestId("workExperienceRecap")).toBeVisible();
  });
});
