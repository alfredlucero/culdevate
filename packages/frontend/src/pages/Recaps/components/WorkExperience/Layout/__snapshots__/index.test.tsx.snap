// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkExperienceLayout /> should render without error 1`] = `
<div>
  <div
    class="extra-work-experience-layout-class"
    data-testid="workExperienceLayoutTestId"
  >
    <header
      class="mb-8"
    >
      <button
        class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mb-4"
        type="button"
      >
        <div
          class="flex items-center"
        >
          <div
            class="font-sans text-2xl mr-2"
            id=""
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-chevron-left fa-w-10 "
              data-icon="chevron-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                fill="currentColor"
              />
            </svg>
          </div>
           Back to Recaps
        </div>
      </button>
      <div>
        <div
          class="flex justify-between w-full"
        >
          <h2
            class="font-mono font-bold text-gray-800 text-3xl "
          >
            Work Experience
          </h2>
          <button
            class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-teal-500 text-white hover:bg-teal-700 "
            type="button"
          >
            <div
              class="flex items-center"
            >
              Add a Recap
            </div>
          </button>
        </div>
        <p
          class="font-sans text-gray-800 text-base "
        >
          Recap everything about your career from internships to full-time jobs and opportunities.
        </p>
      </div>
    </header>
    <section
      class=""
    >
      <div
        class="bg-white shadow-md rounded px-8 py-6 p-4 flex mb-4"
        data-testid="workExperienceRecap"
      >
        <div
          class="flex justify-center items-center p-3 rounded-lg bg-gray-200 recap-icon mr-4"
        >
          <div
            class="font-sans text-3xl text-teal-400"
            id=""
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-suitcase fa-w-16 "
              data-icon="suitcase"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128 480h256V80c0-26.5-21.5-48-48-48H176c-26.5 0-48 21.5-48 48v400zm64-384h128v32H192V96zm320 80v256c0 26.5-21.5 48-48 48h-48V128h48c26.5 0 48 21.5 48 48zM96 480H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48h48v352z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <div
          class="flex-auto"
        >
          <h4
            class="font-mono font-bold text-gray-800 text-xl "
          >
            Lead Software Engineer
          </h4>
          <p
            class="font-sans text-gray-800 text-base "
          >
            Culdevate
             | 
            Self-Employed
          </p>
          <p
            class="font-sans text-gray-800 text-base italic text-gray-600"
          >
            Jan 2020
             -
             
            Present
          </p>
          <p
            class="font-sans text-gray-800 text-base "
          >
            Long Beach, CA
          </p>
          <ul
            class="mt-3"
          >
            <li
              class="flex items-center"
            >
              <div
                class="font-sans text-2xl mr-2 text-teal-300"
                id=""
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-square fa-w-14 "
                  data-icon="square"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <p
                class="font-sans text-gray-800 text-base pt-1"
              >
                Worked on the Culdevate start up seriously
              </p>
            </li>
            <li
              class="flex items-center"
            >
              <div
                class="font-sans text-2xl mr-2 text-teal-300"
                id=""
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-square fa-w-14 "
                  data-icon="square"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <p
                class="font-sans text-gray-800 text-base pt-1"
              >
                Getting better at making REST API endpoints
              </p>
            </li>
            <li
              class="flex items-center"
            >
              <div
                class="font-sans text-2xl mr-2 text-teal-300"
                id=""
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-square fa-w-14 "
                  data-icon="square"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <p
                class="font-sans text-gray-800 text-base pt-1"
              >
                Improving interviewing and mentoring skills
              </p>
            </li>
          </ul>
        </div>
        <div
          class="w-1/3 text-right"
        >
          <button
            class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mr-2"
            id="workExperienceId"
            type="button"
          >
            <div
              class="flex items-center"
            >
              <div
                class="font-sans text-2xl mr-2"
                id=""
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-pencil-alt fa-w-16 "
                  data-icon="pencil-alt"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                    fill="currentColor"
                  />
                </svg>
              </div>
               Edit
            </div>
          </button>
          <button
            class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-red-500 text-white hover:bg-red-700 "
            id="workExperienceId"
            type="button"
          >
            <div
              class="flex items-center"
            >
              <div
                class="font-sans text-2xl mr-2"
                id=""
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-trash-alt fa-w-14 "
                  data-icon="trash-alt"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              Delete
            </div>
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;
